pandas/core/frame.py

```diff
@@ -5545,20 +5545,46 @@ def _replace_columnwise(
     @doc(NDFrame.shift, klass=_shared_doc_kwargs["klass"])
     def shift(
         self,
-        periods: int = 1,
+        periods: int | Iterable[int] = 1,
         freq: Frequency | None = None,
         axis: Axis = 0,
         fill_value: Hashable = lib.no_default,
+        suffix: str | None = None,
```

Comment:
Have we thought about the behavior a user might expect if suffix is passes and periods is an int?

Reply:
Ah, good point. It should probably fail actually?